### PR TITLE
feat: Added Consul example

### PR DESCRIPTION
**Pull request by jcountsNR**
Example for scraping HCP Consul metrics API and sending the metrics to New Relic.

### CODE REVIEW

File: other-examples/collector/hcp-consul/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+docker compose up
+```
+
+## Local Variable information
+
+| Variable | Description | Docs |
+| -------- | ----------- | ---- |
+| **NEW_RELIC_API_KEY** |New Relic Ingest API Key |[API Key docs](https://docs.newrelic.com/docs/apis/intro-apis/new-relic-api-keys/) | 
+| **NEW_RELIC_OTLP_ENDPOINT** |Default US OTLP endpoint is https://otlp.nr-data.net | [OTLP endpoint config docs](https://docs.newrelic.com/docs/more-integrations/open-source-telemetry-integrations/opentelemetry/get-started/opentelemetry-set-up-your-app/#review-settings) |
+| **CONSUL_ACCESS_URL** | URL for communicating with your HCP managed Consul cluster |[Consul URL docs](https://developer.hashicorp.com/hcp/docs/consul/hcp-managed/access#get-access-url)|
```
Comment (jack-berg) [inline other-examples/collector/hcp-consul/README.md:31]: Looks like you changed `CONSOL_*` to `HCP_*`. Can you update this table?

File: other-examples/collector/hcp-consul/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+version: "3.6"
```
Comment (jack-berg) [inline other-examples/collector/hcp-consul/docker-compose.yaml:1]:
```suggestion
version: "3"
```

File: other-examples/collector/hcp-consul/collector.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+          dns_sd_configs:
+            - names:
+                - "$HCP_ACCESS_URL"
+              type: "A"
+              port: 443
+          authorization:
+            credentials: "$HCP_ACCESS_TOKEN"
+          metrics_path: "/v1/agent/metrics"
+          tls_config:
+            insecure_skip_verify: true
```
Comment (jack-berg) [inline other-examples/collector/hcp-consul/collector.yaml:18]: Why do we want to skip tls verification?

File: other-examples/collector/hcp-consul/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# HCP Consul OpenTelemetry metrics example setup
```
Comment (jack-berg) [inline other-examples/collector/hcp-consul/README.md:1]: Can you link to this directly from the root readme?

File: other-examples/collector/hcp-consul/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+This example shows a setup for running a prometheus OpenTelemetry Collector in a docker container to scrape metrics from HCP Consul and post them the New Relic OTLP Collector Endpoint.
+
+**NOTE**: This is *not* for self-managed Consul, and will only work with HCP managed Consul. For self-managed Consul you will have to communicate with the [agent API](https://developer.hashicorp.com/consul/api-docs/agent).
```
Comment (jack-berg) [inline other-examples/collector/hcp-consul/README.md:5]: With the confluent example, we linked out to a page in our docs which explains how to install a dashboard based on these metrics and how to query for these metrics. Is there any equivalent for this?

File: other-examples/collector/hcp-consul/README.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# HCP Consul OpenTelemetry metrics example setup
+
+This example shows a setup for running a prometheus OpenTelemetry Collector in a docker container to scrape metrics from HCP Consul and post them the New Relic OTLP Collector Endpoint.
```
Comment (jack-berg) [inline other-examples/collector/hcp-consul/README.md:3]: Maybe link the "HCP Consul" text to something explaining what HCP consul is?

File: other-examples/collector/hcp-consul/collector.yaml (hunk 3 above)
Comment (jcountsNR) [reply other-examples/collector/hcp-consul/collector.yaml:18]: This is the recommended setting in HCP docs, so it's what I used. I'm not sure if it's required or not, but even their docs for scraping from self-managed API's includes this setting.

https://developer.hashicorp.com/hcp/docs/consul/monitor/metrics#prometheus

Comment (jack-berg) [reply other-examples/collector/hcp-consul/collector.yaml:18]: Can you try running without this option? If required, can you include a comment pointing to these docs for this?

**Pull request merged**